### ISSUE TITLE
Both `block-closing-brace-*-before` rules grow a block ending on a custom property with no semicolon behind it

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#538]
### What minimal example or steps are needed to reproduce the bug?

A custom property keeps its value's trailing whitespace, and where no semicolon closes the declaration it keeps whatever is written behind it as well. So the run standing in front of the closing brace of such a block is the value's own text, and the block's raw is empty:

```js
postcss.parse(`a { --b:red }`)
// the rule: raws.after ``
// the declaration: value `red `
```

Both `block-closing-brace-*-before` rules measure that run — they read the block as the file prints it, down to the brace — and both write into the block's `raws.after`, which stands behind the run they measured. So the write lands past the text the check was about, the next parse hands the whole of it back inside the value, and the raw is empty again.

```css
a { --b:red}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/block-closing-brace-newline-before": "always"
	}
}
```

`block-closing-brace-space-before` under `"always"` reaches it too.

### What did you expect to happen?

The whitespace the rule reports to be the whitespace it writes over, and one run of `--fix` to settle the file.

### What actually happened?

The file grows by a character on every run and never comes to rest. All three syntaxes answer alike:

| rule | option | input | run 1 | run 2 | run 3 |
| --- | --- | --- | --- | --- | --- |
| `block-closing-brace-newline-before` | `always` | `a { --b:red}` | `a { --b:red\n}` | `a { --b:red\n\n}` | `a { --b:red\n\n\n}` |
| `block-closing-brace-newline-before` | `always` | `a { --b:}` | `a { --b:\n}` | `a { --b:\n\n}` | `a { --b:\n\n\n}` |
| `block-closing-brace-space-before` | `always` | `a { --b:\t}` | `a { --b:\t }` | `a { --b:\t  }` | `a { --b:\t   }` |
| `block-closing-brace-space-before` | `always` | `a { --b:red  }` | `a { --b:red   }` | `a { --b:red    }` | `a { --b:red     }` |

The boundary is the declaration and not the shape of its value. Over 36 blocks per rule — a custom property and a plain one, with a semicolon behind the declaration and without, and nine values apiece: nothing at all, one space, two spaces, a tab, a line break, a word, and that word followed by a space, by two spaces and by a tab — 14 grow, and every one of them is a custom property closing its block with no semicolon behind it. A plain property never grows, since the parser hands its trailing whitespace to the block; neither does any declaration a semicolon closes.

Inside that boundary the two rules differ by what each is content with. `block-closing-brace-newline-before` grows on all nine values. `block-closing-brace-space-before` grows on five: two spaces, a tab, a line break, the word followed by two spaces and the word followed by a tab — that is, wherever the value ends in whitespace that is neither nothing nor a lone space, so that the single space it writes lands behind whitespace it then reports again. It rests on the other four, where one write leaves exactly the single space the option asks for.

### Notes

Found by the sweep of [#387](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/387), which reads the same run from the other end and where it is measured on both sides of that branch alike: 2 574 rows of that corpus never come to rest, and every one of them is one of these two rules under `always`.

It is #387's shape wearing the other face. There the rule read a run the declaration no longer holds and wrote into the declaration; here the rule reads a run the declaration does hold and writes into the block. Both are one rule measuring one text and writing another, and the cure is the same: write where the run stands. `getBlockAfter` and `setBlockAfter` already name the question of which raw holds a block's final run, and this is the case they do not cover — the run is in no raw of the block at all.

[#374](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/374) is the same pair of rules never settling on a Less mixin call carrying `!important`, and it is a different cause: there the run belongs to neither raw because the parser collects both sides of the flag into one.
